src/Interfaces/AbstractZonotope.jl

```diff
@@ -69,7 +69,11 @@ function genmat_fallback(Z::AbstractZonotope{N}) where {N<:Real}
     if isempty(gens)
         return Matrix{N}(undef, dim(Z), 0)
     end
-    return hcat(gens...)
+    G = Matrix{N}(undef, dim(Z), length(gens))
+    @inbounds for (i, g) in enumerate(gens)
```

Comment:
I realized only now that `length(gens)` is not good because that creates the generators just for computing the length and then creates them another time in the loop. I have to look into `push!` for matrices.

Reply:
> that creates the generators just for computing the length

Why? Isn't `Base.length(it::FallbackGeneratorIterator) = it.n_plus_one - 1` what is used in the generic fallback?

Reply:
True, but there can be other implementations apart from `FallbackGeneratorIterator` that dynamically compute the generators. The `HyperrectangleGeneratorIterator` goes a hybrid way by precomputing the `length` and storing it, but I am not sure that this is smart. We might change this in the future.

https://github.com/JuliaReach/LazySets.jl/blob/c61beeacc7ad7eae8634a03f8e2fb7aef65bb8c9/src/Interfaces/AbstractHyperrectangle.jl#L64-L98

Reply:
Ok, i see. Then you could be looking to `push!` to a vector and then [reshape](https://docs.julialang.org/en/v1/base/arrays/#Base.reshape) it.

Reply:
After benchmarking the generic implementation, I decided to have both versions. The generic fallback with an additional `Vector` is used by default but currently all our types can efficiently determine the number of generators and so I let them call the more efficient version.
```julia
julia> @btime genmat(rand(Interval));  # knowing the generators
  179.210 ns (4 allocations: 272 bytes)

julia> @btime genmat(rand(Interval));  # not knowing the generators
  309.431 ns (7 allocations: 496 bytes)

julia> @btime genmat(rand(BallInf, dim=100));  # knowing the generators
  25.119 μs (7 allocations: 80.02 KiB)

julia> @btime genmat(rand(BallInf, dim=100));  # not knowing the generators
  26.991 μs (114 allocations: 169.72 KiB)
```
For `Interval`s I think we can do even better by implementing a custom `genmat` method (but not here).